packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/creditCard.js

```diff
@@ -122,6 +116,29 @@ const CreditCard = props => {
         }
     }, [errors, classes.error, classes.errors_container]);
 
+    /**
+     * These 2 functions are wrappers around the `isRequired` function
+     * of `formValidators`. They perform validations only if the
+     * billing address is different from shipping address.
+     *
+     * We write this function in `venia-ui` and not in the `peregrine` talon
+     * because it references `isRequired` which is a `venia-ui` function.
+     */
+    const isFieldRequired = useCallback(
+        value => {
+            if (isBillingAddressSame) {
+                /**
+                 * Informed validator functions return `undefined` is
+                 * validation is `true`
+                 */
+                return undefined;
+            } else {
+                return isRequired(value);
+            }
+        },
+        [isBillingAddressSame]
+    );
```

Comment:
Why is this form being submitted if `isBillingAddressSame` is `true`? Do we need to submit anything here? I would think if the billing and shipping address are the same (as in the user clicked the checkbox) then we would have sent that information via the graphQL cart mutation which means that this form shouldn't even render.

Reply:
The billing address fields are always rendered, they are just hidden if the checkbox is checked. This was due to a review comment way up on the top 😅 . Hence when I have written a wrapper around the `isRequired` to only validate if the boolean is false.